### ISSUE TITLE
Resources are not re-resolved after installing extensions

### ISSUE THREAD

**Issue by alexweininger** [microsoft/vscode-azureresourcegroups#280]
Shallow resources that say "Install extension to enable additional features..." are not re-resolved and refreshed upon installing an extension. The workaround is to reload VS Code 😞

**Comment by alexweininger** [comment]
> It seems like we really need to dump the cache on refresh.

Yep, that's the issue. I experimented with checking to see if the resolver returned by `getResolver()` has changed since the last resolve, and clearing the cache if it has. And it seems to work well.

**Comment by bwateratmsft** [comment]
I mean, the _whole_ cache. We keep encountering issues because we're trying too hard to cache things. I don't see why we should keep anything between refreshes (except the ID of focused nodes).
